autoscheduler/frontend/src/redux/reducers.ts

```diff
@@ -149,9 +149,8 @@ function availability(
                 endTimeHours: newEndHrs,
                 endTimeMinutes: newEndMins,
               };
-              if (overlapsFound >= 2) {
+              if (overlapsFound === 2) {
```

Comment:
This was actually a very good comment to make. Originally, when an overlap occurred, the merged availability would be inserted at the index of the pre-existing availability that the new one overlapped with. However, this would insert the availability multiple times if the new availability overlapped with multiple pre-existing availabilities, leading to the bug that Ryan reported. The splicing was how I got rid of the duplicates. Originally, I had assumed that it would not be possible for more than 2 overlaps to occur, so I didn't allow splicing for that, but after Ryan's bug report, I realized I might have to splice more often than anticipated. That is roughly why I changed it from `=== 2` to `>= 2`. However, this led to too much splicing, accidentally deleting unrelated availabilities. To fix this, I removed splicing altogether, and instead, I just insert the new, merged availability at the very end. Doing so eliminates the need for splicing altogether, and also simplifies the code somewhat.